plugins/chat/spec/system/list_channels/no_sidebar_spec.rb

```diff
@@ -123,4 +123,21 @@
       expect(page).to have_no_css(".direct-message-channels-section")
     end
   end
+
+  context "when public channels are disabled" do
+    before { SiteSetting.enable_public_channels = false }
+
+    it "shows the create direct message button" do
+      visit("/chat")
+
+      expect(chat).to have_direct_message_channels_section
+    end
+
+    it "shows the create direct message button in the drawer" do
```

Comment:
I wonder if we should explicitly enable chat drawer here with `chat_page.prefers_drawer` either in the test or before block?